yt_dlp/extractor/pialive.py

```diff
@@ -0,0 +1,127 @@
+            },
+            'params': {
+                'getcomments': True,
+                'skip_download': True,
+                'ignore_no_formats_error': True,
+            },
+            'skip': 'The video is no longer available',
+        },
+    ]
+
+    def _extract_vars(self, variable, html):
+        return self._search_regex(
+            rf'(?:var|const|let)\s+{variable}\s*=\s*(["\'])(?P<value>(?:(?!\1).)+)\1',
+            html, f'variable {variable}', group='value')
+
+    def _real_extract(self, url):
+        video_key = self._match_id(url)
+        webpage = self._download_webpage(url, video_key)
+
+        program_code = self._extract_vars('programCode', webpage)
+        article_code = self._extract_vars('articleCode', webpage)
+        title = self._html_extract_title(webpage)
+
+        if get_element_html_by_class('play-end', webpage):
+            raise ExtractorError('The video is no longer available', expected=True, video_id=program_code)
+
+        if start_info := clean_html(get_element_by_class('play-waiting__date', webpage)):
+            date, time = self._search_regex(
+                r'(?P<date>\d{4}/\d{1,2}/\d{1,2})\([月火水木金土日]\)(?P<time>\d{2}:\d{2})',
+                start_info, 'start_info', fatal=False, group=('date', 'time'))
+            if all((date, time)):
+                release_timestamp_str = f'{date} {time} +09:00'
+                release_timestamp = unified_timestamp(release_timestamp_str)
+                self.raise_no_formats(f'The video will be available after {release_timestamp_str}', expected=True)
+                return {
+                    'id': program_code,
+                    'title': title,
+                    'live_status': 'is_upcoming',
+                    'release_timestamp': release_timestamp,
+                }
+
+        payload, content_type = multipart_encode({
+            'play_url': video_key,
+            'api_key': self.API_KEY,
+        })
+        api_kwargs = {
+            'video_id': program_code,
+            'data': payload,
+            'headers': {'Content-Type': content_type, 'Referer': self.PLAYER_ROOT_URL},
+        }
+
+        player_tag_list = self._download_json(
+            f'{self.PIA_LIVE_API_URL}/perf/player-tag-list/{program_code}', **api_kwargs,
+            note='Fetching player tag list', errnote='Unable to fetch player tag list')
+        if self.get_param('getcomments'):
+            chat_room_url = traverse_obj(self._download_json(
+                f'{self.PIA_LIVE_API_URL}/perf/chat-tag-list/{program_code}/{article_code}', **api_kwargs,
+                note='Fetching chat info', errnote='Unable to fetch chat info', fatal=False),
+                ('data', 'chat_one_tag', {extract_attributes}, 'src', {url_or_none}))
+        else:
+            chat_room_url = None
+
+        return self.url_result(
+            extract_attributes(player_tag_list['data']['movie_one_tag'])['src'], url_transparent=True,
```

Comment:
Can the returned URL be for different sites, or is this also always `PIAULIZAPortalAPIIE`?

Reply:
> Can the returned URL be for different sites, or is this also always `PIAULIZAPortalAPIIE`?

It's not always `PIAULIZAPortalAPIIE`.

> This website sells tickets for 5 streaming platforms ( [`ULIZA`](https://uliza.jp/), [`VIMEO`](https://vimeo.com/), [`Smart STREAM`](https://www.smartstream.ne.jp/), [`Theater ZA`](https://za.theater/), [`Live Extreme`](https://www.live-extreme.net/) ) ( https://pia-live.jp/requirement?lang=en ).

.